### PR TITLE
Added the package and publish Helm chart workflow

### PR DESCRIPTION
**Pull request by lnowotny**
### What does this Pull Request accomplish?

SystemLink has many services in GitHub repos that would like to share certain parts of the build process. This PR introduces a reusable workflow for the packaging and publishing of Helm charts.

### Why should this Pull Request be merged?

The new workflow is needed to reduce duplicated logic in our build workflows.

### What testing has been done?

I updated a helium service to use this workflow (locally within the repo) and the workflow succeeded:
https://github.com/ni-kismet/helium-security-ui/actions/runs/2761645106

### CODE REVIEW

**Review by BKnight760 — APPROVED** [review]

File: bin/package-and-publish-helm-chart.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        break
+
+  # Parse version from the Chart.yaml
+  chart_version = '0.0.0'
+  with open(chart_filename) as file:
+    for line in file:
+      if line.startswith('version:'):
+        chart_version = line.split(':')[1].strip()
+        print(f'Chart version: {chart_version}')
+        break
```
Comment (irwand) [inline bin/package-and-publish-helm-chart.py]: these 2 are similar, maybe use a function to limit duplication?

File: bin/package-and-publish-helm-chart.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+  version_elements = chart_version.split('.')
+  version_major = int(version_elements[0])
+  version_minor = int(version_elements[1])
+  version_patch = int(version_elements[2])
```
Comment (irwand) [inline bin/package-and-publish-helm-chart.py]:
```suggestion
  version_major, version_minor, version_patch = map(int, chart_version.split('.'))
```
This logic is repeated below, so you might make it another function, maybe?

Comment (irwand) [reply bin/package-and-publish-helm-chart.py]: oh, except that it won't work if chart_version is something like 1.2.3.dev4  .. in that case you might add [:3] in there to limit the split() to only 3 elements

File: bin/package-and-publish-helm-chart.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        groups = result.groups()
+
+        package_version = groups[1]
+        print(f'Package version: {package_version}')
+        package_version_elements = package_version.split('.')
+        package_major = int(package_version_elements[0])
+        package_minor = int(package_version_elements[1])
+        package_patch = int(package_version_elements[2])
+
+        if ((package_major >= highest_version[0]) and (package_minor >= highest_version[1]) and (package_patch > highest_version[2])):
```
Comment (irwand) [inline bin/package-and-publish-helm-chart.py]: not sure if this is intended .. that seems to only work to compare 5.5.5 vs 4.4.4 but won't work when comparing 5.4.1 vs 4.4.4? 
One easy way to compare might be to stringify it with 0 padding, something like 5.4.1 becomes 050401, which is higher than 040404 ..
but now it becomes a little more involved, you might want to create a class Version to be a bit cleaner. 
Not sure if you can just use [niversion](https://dev.azure.com/ni/DevCentral/_git/Tools?path=/shared/niversion/niversion/niversion.py).BaseVersion

**Review by irwand — APPROVED** [review]

File: bin/package-and-publish-helm-chart.py (hunk 1 above)
Comment (lnowotny) [reply bin/package-and-publish-helm-chart.py]: The script is essentially just getting moved in this PR. That being said, I can look into this feedback.

File: bin/package-and-publish-helm-chart.py (hunk 3 above)
Comment (neilvana) [reply bin/package-and-publish-helm-chart.py]: Alternatively I think you can use the very lightweight `packaging` library (but you will have to add a `pip install packaging` before using it)

```python
>>> from packaging import version
>>> version.parse("5.4.1") > version.parse("4.4.4")
True
```

**Review by neilvana — APPROVED** [review]

File: bin/package-and-publish-helm-chart.py (hunk 3 above)
Comment (lnowotny) [reply bin/package-and-publish-helm-chart.py]: This was a good catch. I'm working on getting a fix submitted to the AzDo version of this script and then I will push an update here.

**Pull request merged**